lib/improver/psychrometric_calculations/psychrometric_calculations.py

```diff
@@ -415,49 +403,79 @@ def calculate_wet_bulb_temperature(self, temperature, relative_humidity,
                 Cube of wet bulb temperature (K).
 
         """
-        precision = np.full(temperature.data.shape, self.precision)
-
-        # Set units of input diagnostics.
         relative_humidity.convert_units(1)
         pressure.convert_units('Pa')
         temperature.convert_units('K')
 
+        precision = np.full(temperature.data.shape, self.precision)
+        wbt = temperature.copy()
+        wbt.rename('wet_bulb_temperature')
+
+        latent_heat = Utilities.latent_heat_of_condensation(temperature)
```

Comment:
`latent_heat_of_condensation` could also work in arrays

Reply:
Due to the converting it to celsius, this is where major errors arose.